### PR TITLE
Object Adapter cleanup

### PR DESCRIPTION
**Pull request by bernardnormier**
Various cleanup of ObjectAdapter and related code.

### CODE REVIEW

File: csharp/src/Ice/ObjectAdapter.cs [hunk 1]
```diff
@@ -920,387 +861,289 @@ public void SetPublishedEndpoints(IEndpoint[] newEndpoints)
...
             try
             {
-                int threadPoolSize = communicator.GetPropertyAsInt($"{_name}.ThreadPool.Size") ?? 0;
-                int threadPoolSizeMax = communicator.GetPropertyAsInt($"{_name}.ThreadPool.SizeMax") ?? 0;
+                int threadPoolSize = Communicator.GetPropertyAsInt($"{Name}.ThreadPool.Size") ?? 0;
+                int threadPoolSizeMax = Communicator.GetPropertyAsInt($"{Name}.ThreadPool.SizeMax") ?? 0;
                 if (threadPoolSize > 0 || threadPoolSizeMax > 0)
                 {
-                    _threadPool = new ThreadPool(_communicator, _name + ".ThreadPool", 0);
+                    _threadPool = new ThreadPool(Communicator, Name + ".ThreadPool", 0);
```
Comment (pepone) [inline csharp/src/Ice/ObjectAdapter.cs]: replace with `"${Name}.ThreadPool"`

File: csharp/src/Ice/ObjectAdapter.cs [hunk 2]
```diff
@@ -920,387 +861,289 @@ public void SetPublishedEndpoints(IEndpoint[] newEndpoints)
...
                     }
                 }
 
-                //
                 // Parse published endpoints.
-                //
                 _publishedEndpoints = ComputePublishedEndpoints();
-                ILocatorPrx? locator = communicator.GetPropertyAsProxy($"{_name}.Locator", ILocatorPrx.Factory);
-                if (locator != null)
+                ILocatorPrx? locator = Communicator.GetPropertyAsProxy($"{Name}.Locator", ILocatorPrx.Factory);
```
Comment (pepone) [inline csharp/src/Ice/ObjectAdapter.cs]: maybe remove this intermediary variable 
```
Locator = Communicator.GetPropertyAsProxy($"{Name}.Locator", ILocatorPrx.Factory) ??
                    Communicator.GetDefaultLocator();
```

File: csharp/src/Ice/ObjectAdapter.cs [hunk 3]
```diff
@@ -920,387 +861,289 @@ public void SetPublishedEndpoints(IEndpoint[] newEndpoints)
...
+                    }
+                    return false;
                 }
             }
-            catch (LocalException)
+        }
+
+        internal void UpdateConnectionObservers()
+        {
+            IncomingConnectionFactory[] factories = Array.Empty<IncomingConnectionFactory>();
```
Comment (pepone) [inline csharp/src/Ice/ObjectAdapter.cs]: avoid initialize it here

File: csharp/src/Ice/Communicator-ObjectAdapterFactory.cs [hunk 4]
```diff
@@ -2,69 +2,202 @@
...
     public sealed partial class Communicator
     {
-        internal ObjectAdapter CreateObjectAdapter(string name, IRouterPrx? router)
+        /// <summary>Shuts down this communicator's server functionality. This triggers the deactivation of all
+        /// object adapters. After this method returns, no new requests are processed. However, requests that have
+        /// been started before Shutdown was called might still be active. You can use <see cref="WaitForShutdown"/>
+        /// to wait for the completion of all requests.</summary>
+        public void Shutdown()
         {
+            ObjectAdapter[] adapters = Array.Empty<ObjectAdapter>();
```
[the remark this answers is not in the document]
Comment (bernardnormier) [reply csharp/src/Ice/Communicator-ObjectAdapterFactory.cs]: Fixed, thanks.

**Pull request merged**